app/models/manageiq/providers/redhat/infra_manager/provision_via_pxe/state_machine.rb

```diff
@@ -3,6 +3,19 @@ def configure_destination
     signal :create_pxe_configuration_file
   end
 
+  def customize_destination
+    if destination_image_locked?
+      _log.info("Destination image locked; re-queuing")
+      requeue_phase
+    else
+      message = "Starting New #{destination_type} Customization"
+      _log.info("#{message} #{for_destination}")
+      update_and_notify_parent(:message => message)
+      configure_container
+      signal :create_pxe_configuration_file
```

Comment:
I think the base `customize_destination` should signal a new state called `customize_guest` and this new state should be responsible for configuring cloudinit, pxe or ISO.